jetty-http2/http2-server/src/main/java/org/eclipse/jetty/http2/server/HttpTransportOverHTTP2.java

```diff
@@ -110,6 +111,13 @@ public void send(MetaData.Response info, boolean isHeadRequest, ByteBuffer conte
             {
                 if (commit.compareAndSet(false, true))
                 {
+                    if (lastContent)
+                    {
+                        HttpFields responseHeaders = info.getFields();
+                        if (!responseHeaders.contains(HttpHeader.CONTENT_LENGTH))
+                            responseHeaders.put(HttpHeader.CONTENT_LENGTH, String.valueOf(BufferUtil.length(content)));
```

Comment:
Does it have to be put into the `HttpFields` instance?  The HTTP/1 does this in the generator, so that as it is committing the headers it looks for an existing content-length header and if one is not found, but the length is known, then the content length header is generated directly into the output buffer without creating a `HttpField` instance or modifying the info.

The h2 approach here still results in different behaviour because the HttpFields class will be different after commit.

Reply:
@gregw HTTP/2 works with frames, so if I can't put the `Content-Length` inside the `HttpFields` I would need an additional field `HeadersFrame.contentLength` which is doable but feels weird.

Also, it's not clear what should be done in case of conflicts with the application.
For example, if the application specified a different `Content-Length` value?